webpack/assets/javascripts/react_app/redux/API/APIHelpers.js

```diff
@@ -0,0 +1,75 @@
+import { STATUS } from '../../constants';
+
+export const formatSuccess = ({
+  data = {},
+  payload = {},
+  status = STATUS.RESOLVED,
+  error = null,
+}) => ({
+  status,
+  error,
+  ...payload,
+  ...data,
+});
+
+export const formatFailure = ({
+  error,
+  payload = {},
+  status = STATUS.ERROR,
+}) => ({
+  status,
+  error,
+  ...payload,
+});
+
+export const formatPending = ({
+  payload = {},
+  status = STATUS.PENDING,
+  error = null,
+}) => ({
+  status,
+  error,
+  ...payload,
+});
+
+export const formatSuccessByPayloadKey = (response, payloadKey) => ({
+  [response.payload[payloadKey]]: formatSuccess(response),
```

Comment:
I'm wondering if we actually need the `formatById`, would it be easier to implement `manageById` as many different API calls? E.g. have a different key for each.

Reply:
IMHO it feels better having each component scope separately in the store, e.g.:
```js
{
  API : {
    BOOKMARKS: { ... },
    FACT_CHART: { ... },
    POWER_STATUS: { ... }
    AUTOCOMPLETE: { ... }
  }
}
```
instead of a huge list of different API keys, and in this example I am showing only the tip of it:
I think we should keep the store more organized.
```js
{
  API : {
    BOOKMARKS_1: { ... },
    BOOKMARKS_35: { ... },
    FACT_CHART_53: { ... },
    FACT_CHART_227: { ... },
    FACT_CHART_9: { ... },
    FACT_CHART_45: { ... },
    FACT_CHART_5: { ... },
    POWER_STATUS_2: { ... },
    POWER_STATUS_5: { ... },
    POWER_STATUS_14: { ... },
    AUTOCOMPLETE_1: { ... },
    AUTOCOMPLETE_72: { ... }
  }
}
```

Reply:
The store is a big chunk of data, doesn't need to be organized, for that we have selectors which create the logic order, so why  "paying" on building and maintaining such a structure if we can memorize the retrieval with selectors?